broker/src/serve_tasks.rs

```diff
@@ -87,8 +87,58 @@ async fn get_results_for_task(
     Ok((statuscode, Json(results)))
 }
 
-fn would_wait_for_elements<S>(vec: &Vec<S>, block: &HowLongToBlock) -> bool {
-    usize::from(block.wait_count.unwrap_or(0)) > vec.len()
+// GET /v1/tasks/:task_id/results/stream
+async fn get_results_for_task_stream(
+    ConnectInfo(addr): ConnectInfo<SocketAddr>,
+    State(state): State<TasksState>,
+    block: HowLongToBlock,
+    task_id: MsgId,
+    msg: MsgSigned<MsgEmpty>
+) -> Result<Sse<impl Stream<Item = Result<Event, Infallible>>>,(StatusCode, &'static str)> {
+    debug!("get_results_for_task_stream(task={}) called by {} with IP {addr}, wait={:?}", task_id.to_string(), msg.get_from(), block);
+    let (mut results, rx_new_result, rx_deleted_task)  = {
+        let tasks = state.tasks.read().await;
+        let Some(task) = tasks.get(&task_id) else {
+            return Err((StatusCode::NOT_FOUND, "Task not found"));
+        };
+        if task.get_from() != msg.get_from() {
+            return Err((StatusCode::UNAUTHORIZED, "Not your task."));
+        }
+        let results = task.msg.results.clone();
+        let rx_new_result = match would_wait_for_elements(results.len(), &block) {
+            true => Some(state.new_result_tx.read().await.get(&task_id)
+                .unwrap_or_else(|| panic!("Internal error: No new_result_tx found for task {}", task_id))
+                .subscribe()),
+            false => None,
+        };
+        (results, rx_new_result, state.removed_task_rx.subscribe())
+    };
+
+    let stream = async_stream::stream! {
+        for (_from, result) in &results {
+            let event = Ok(Event::default()
+                .event(SseEventType::NewResult)
+                .json_data(result)
+                .unwrap());
```

Comment:
This probably includes allocation, so failing is possible. If possible, please handle the error.

Reply:
Fixed in 29150a79cc6ff59fb1defcab2df62475b152fe52